cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -532,23 +379,29 @@ func (cfg *Config) validate() error {
 		return fmt.Errorf("subscription ID not set")
 	}
 
-	if cfg.UseManagedIdentityExtension {
-		return nil
+	if cfg.UseManagedIdentityExtension && cfg.UseFederatedWorkloadIdentityExtension {
+		return fmt.Errorf("you can not combine both managed identity and workload identity as an authentication mechanism")
 	}
 
-	if cfg.TenantID == "" {
-		return fmt.Errorf("tenant ID not set")
+	if cfg.VMType != providerazureconsts.VMTypeStandard && cfg.VMType != providerazureconsts.VMTypeVMSS {
+		return fmt.Errorf("unsupported VM type: %s", cfg.VMType)
 	}
```

Comment:
The comment on VMType still says vmssflex value is supported. Also need to think through migration, if that's indeed how it was intended to be set, and is changing now

Reply:
```go
func (m *AzureManager) buildNodeGroupFromSpec(spec string) (cloudprovider.NodeGroup, error) {
	scaleToZeroSupported := scaleToZeroSupportedStandard
	if strings.EqualFold(m.config.VMType, providerazureconsts.VMTypeVMSS) {
		scaleToZeroSupported = scaleToZeroSupportedVMSS
	}
	s, err := dynamic.SpecFromString(spec, scaleToZeroSupported)
	if err != nil {
		return nil, fmt.Errorf("failed to parse node group spec: %v", err)
	}
	vmsPoolSet := m.azureCache.getVMsPoolSet()
	if _, ok := vmsPoolSet[s.Name]; ok {
		return NewVMsPool(s, m), nil
	}

	switch m.config.VMType {
	case providerazureconsts.VMTypeStandard:
		return NewAgentPool(s, m)
	case providerazureconsts.VMTypeVMSS:
		return NewScaleSet(s, m, -1, false)
	default:
		return nil, fmt.Errorf("vmtype %s not supported", m.config.VMType)
	}
}
```
From this (that has always been here), VM types other than these two will effectively disable CAS already, given this method is critical. I don't think anyone can use that right now.
I think it is fair to "notify users" through fail initialization like this, given the config is cluster-level.

For our support of flex, I think we do have it (that `EnableVMSSFlexNodes`?), but not through this. This sounds like something to revisit.